net/dns/resolver/tsdns.go

```diff
@@ -725,6 +725,22 @@ func (r *Resolver) resolveLocalReverse(name dnsname.FQDN) (dnsname.FQDN, dns.RCo
 		return "", dns.RCodeRefused
 	}
 
+	r.mu.Lock()
+	defer r.mu.Unlock()
+
+	// If the requested IP is part of the IPv6 4-to-6 range, it might
+	// correspond to an IPv4 address (assuming IPv4 is enabled).
+	if ip4, ok := tsaddr.Tailscale6to4(ip); ok {
+		fqdn, code := r.fqdnForIP(ip4, name)
+		if code == dns.RCodeSuccess {
+			return fqdn, code
+		}
+	}
+	return r.fqdnForIP(ip, name)
+}
+
+// r.mu must be held.
+func (r *Resolver) fqdnForIP(ip netaddr.IP, name dnsname.FQDN) (dnsname.FQDN, dns.RCode) {
```

Comment:
our convention for mutex-must-be-held methods is to suffix their name with "Locked"

so `fqdnForIPLocked`. But the comment can also stay.